ADR/nnnn-trusted-task-model.md

```diff
@@ -0,0 +1,238 @@
+to allow/deny sets of Tasks based on some criteria.
+
+If the current logic for trusting Tasks is as follows (pseudocode):
+
+```text
+is_on_trusted_tasks_list(task_reference)
+and not is_expired(task_reference)
+```
+
+Then, with the addition of `trusted_task_rules`, it becomes:
+
+```text
+(
+  is_on_trusted_tasks_list(task_reference)
+  or is_allowed_by_trusted_task_rules(task_reference)
+)
+and not is_expired(task_reference)
+and not is_denied_by_trusted_task_rules(task_reference)
+```
+
+Note:
+
+- Both the `trusted_tasks` and the `trusted_task_rules` mechanisms should be supported,
+  at least until decentralization is complete
+- Denial takes priority
+  - Even if a Task is on the `trusted_tasks` list, a deny rule should be able to
+    make it untrusted.
+  - Even if an allow rule makes a Task trusted, if it's expired according to the
+    `trusted_tasks` list, it should be considered untrusted.
+
+This ADR doesn't mandate a specific format for `trusted_task_rules`, but does
+come with an example of what it could look like - see the [Appendix](#appendix).
+
+### Trust upstream Konflux Tasks based on bundle URL
+
+A policy author who wants to trust all the upstream Konflux Tasks should do so by
+defining a rule that allows all task bundles from `quay.io/konflux-ci/tekton-catalog/*`
+with no further conditions. The `trusted_task_rules` mechanism must support this.
+
+### Start signing upstream Konflux tasks with a common release key
+
+Trusting all task bundles from `quay.io/konflux-ci/tekton-catalog/*` is not a strong
+trust model (but isn't weaker than the current one - see [Consequences](#consequences)).
+To let policy authors define a stronger trust model, we should aim to start signing
+the upstream Konflux task bundles with a common release key and extend the
+`trusted_task_rules` mechanism to support signature verification.
+
+## Consequences
+
+### `data-acceptable-bundles` no longer blocks decentralization
+
+Decentralized Konflux Tasks are trusted implicitly, as long as they release to the
+right location.
+
+### Strength of trust model for upstream Konflux Tasks
+
+Current trust model, in theory:
+
+- We place trust in the `quay.io/konflux-ci/tekton-catalog/data-acceptable-bundles`
+  artifact
+  - Only those who can write to this artifact can make Tasks trusted
+
+Current trust model, in practice:
+
+- Who can write to `quay.io/konflux-ci/tekton-catalog/data-acceptable-bundles`:
+  - Release pipelines in the Red Hat deployment of Konflux
+  - The CI pipelines for build-definitions
+  - The CI pipelines for other repos that happen to use the same OpenShift namespace
+    as build-definitions
+
+The proposed trust model, without signing key verification:
+
+- We implicitly trust all tasks in `quay.io/konflux-ci/tekton-catalog/*`
+- We trust anyone who can write to any of those repositories
+  - Which is the same list as for `data-acceptable-bundles`
```

Comment:
I was going to object to this but you're right. There are folks that can already update the list of trusted tasks today outside of CI. So this wouldn't be a different trust model.